### PR TITLE
[cxx-interop] Add new tests for consuming shared foreign reference types

### PR DESCRIPTION
**Pull request by Xazax-hun**
Check semantics for functions creating FRTs, and wrapping FRTs.

### CODE REVIEW

**Comment by Xazax-hun** [comment]
@swift-ci please smoke test

**Comment by Xazax-hun** [comment]
Forgot to check what the failure was and now the results are gone, so rerunning. 

@swift-ci please smoke test

**Comment by Xazax-hun** [comment]
@swift-ci please smoke test

**Comment by Xazax-hun** [comment]
@swift-ci please smoke test

File: test/Interop/CxxToSwiftToCxx/consuming-cxx-struct-parameter-back-to-cxx-execution.cpp [hunk 1]
```diff
@@ -160,6 +180,31 @@ int main() {
...
+    // CHECK-NEXT: releaseShared
+  }
+  {
+    SharedFRT sfrt;
+    sfrt.x = 4;
+    auto wrapper = UseCxx::ValueWrapper::init(&sfrt);
+    UseCxx::genericConsumingFunc(wrapper);
+    // CHECK-NEXT: retainShared
+    // CHECK-NEXT: retainShared
+    // CHECK-DAG: releaseShared
```
Comment (Xazax-hun) [inline test/Interop/CxxToSwiftToCxx/consuming-cxx-struct-parameter-back-to-cxx-execution.cpp:205]: These two lines are swapped on linux and windows, this is why I use `CHECK-DAG`.

**Review by egorzhdan — APPROVED** [review]
I haven't verified the expected test output, but thanks for adding these tests!

**Pull request merged**